File: app/src/main/java/sempati/star/app/models/Kelas_armada.java
package sempati.star.app.models;

public class Kelas_armada {
    private float id;
    private String nama_kelas;
    private float jenis_seat;
    private String jumlah_seat;


    // Getter Methods

    public float getId() {
        return id;
    }

    public String getNama_kelas() {
        return nama_kelas;
    }

    public float getJenis_seat() {
        return jenis_seat;
    }

    public String getJumlah_seat() {
        return jumlah_seat;
    }

    // Setter Methods

    public void setId(float id) {
        this.id = id;
    }

    public void setNama_kelas(String nama_kelas) {
        this.nama_kelas = nama_kelas;
    }

    public void setJenis_seat(float jenis_seat) {
        this.jenis_seat = jenis_seat;
    }

    public void setJumlah_seat(String jumlah_seat) {
        this.jumlah_seat = jumlah_seat;
    }
}

File: lib/model/kelasarmada.dart
import 'dart:convert';

class KelasArmada {
  int jenisSeat;
  String namaKelas;
  String jumlahSeat;
  int id;

  KelasArmada({
  this.jenisSeat = 0,
  this.namaKelas = "",
  this.jumlahSeat = "",
  this.id = 0,
  });

  KelasArmada.fromJson(Map<String, dynamic>  map) :
        jenisSeat = map['jenis_seat']  ?? 0,
        namaKelas = map['nama_kelas']  ?? "",
        jumlahSeat = map['jumlah_seat']  ?? "",
        id = map['id']  ?? 0;

  Map<String, dynamic> toJson() => {
        'jenis_seat': jenisSeat,
        'nama_kelas': namaKelas,
        'jumlah_seat': jumlahSeat,
        'id': id,
      };

  KelasArmada copyWith({
    int jenisSeat,
    String namaKelas,
    String jumlahSeat,
    int id,
  }) {
    return KelasArmada(
      jenisSeat: jenisSeat ?? this.jenisSeat,
      namaKelas: namaKelas ?? this.namaKelas,
      jumlahSeat: jumlahSeat ?? this.jumlahSeat,
      id: id ?? this.id,
    );
  }
}

